### PR TITLE
feature/mx-1381 update mex-common to 0.21

### PR DESCRIPTION
**Pull request by cutoffthetop**
# PR Context
- establish compatibility to https://github.com/robert-koch-institut/mex-common/pull/69

# Changes

- update cruft template with new linters
- utilize new, more precise Identifier subclasses
- make pyodbc a soft dependency (only pipelines that use it may fail)

# Removed

- remove sync-persons pipeline, stopgap mx-1572
- remove `public` as a valid sink option

# Fixed

- fix some docstring indents and typings

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,15 +11,15 @@ packages = [{ include = "mex" }]
...
 beautifulsoup4 = "^4.12.3"
-dagster = "^1.6.3"
-dagster-postgres = "^0.22.3"
-dagster-webserver = "^1.6.3"
+dagster = "^1.6.6"
+dagster-postgres = "^0.22.6"
+dagster-webserver = "^1.6.6"
 faker = "^23.0.0"
-mex-common = { git = "https://github.com/robert-koch-institut/mex-common.git", rev = "0.19.3" }
+mex-common = { git = "https://github.com/robert-koch-institut/mex-common.git", rev = "feature/mx-1381-prep-rule-endpoint" }
```
Comment (erichesse) [inline pyproject.toml:18]: reminder to switch to main version again before merging

Comment (cutoffthetop) [reply pyproject.toml]: 1f19a90

File: mex/ifsg/connector.py [hunk 2]
```diff
@@ -17,6 +16,21 @@
...
+
+    def connect(self, _: str) -> None:  # pragma: no cover
+        """Create a new ODBC connection to a database."""
+        return
+
+
+try:
+    import pyodbc  # type: ignore[import-not-found]
+except ImportError:
+    pyodbc = NoOpPyodbc
```
[the remark this answers is not in the document]
Comment (cutoffthetop) [reply mex/ifsg/connector.py:31]: yep good idea, but as discussed in the refinement, we'll do it as part of grippe-web x-system

**Comment by cutoffthetop** [comment]
> And a side note: I forgot to enable the local.env and some tests for `blueant`, `datscha-web` and `confluence-vvt` failed. Probably not an issue for this PR, but we also want to isolate these extractors locally from mex-assets, right?

good point, i created a ticket for this: mx-1575

**Review by erichesse — APPROVED** [review]
Great!

**Pull request merged**